app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
+    unless logged_in_as_admin?
```

Comment:
I've found that a good rule of thumb for "model vs controller" is to ask the question "Would a full-blown API need access to this method?". If it's something that is only ever going to be relevant to the UI, then put it in the controller. Otherwise, put it in the model or a helper that an API controller can also get at.

I also agree that separate methods can help readability and granularity. Any modern compiler or interpreter worth its salt will inline them if required, so always go for human readability and potential reuse if it make sense.